hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/standalone/TransactionExecutors.java

```diff
@@ -67,21 +72,147 @@ public interface TracerBinding extends Supplier<List<OperationTracer>> {
         void runWhere(@NonNull List<OperationTracer> tracers, @NonNull Runnable runnable);
     }
 
+    /**
+     * The properties to use when creating a new {@link TransactionExecutor}.
+     * @param state the {@link State} to use
+     * @param appProperties the properties to use
+     * @param customTracerBinding the custom tracer binding to use
+     * @param customOps the custom operations to use
+     */
+    public record Properties(
+            @NonNull State state,
+            @NonNull Map<String, String> appProperties,
+            @Nullable TracerBinding customTracerBinding,
+            @NonNull Set<Operation> customOps) {
+        /**
+         * Create a new {@link Builder} instance.
+         * @return a new {@link Builder} instance
+         */
+        public static Builder newBuilder() {
+            return new Builder();
+        }
+
+        /**
+         * Builder for {@link Properties}.
+         */
+        public static class Builder {
+            private State state;
+            private TracerBinding customTracerBinding;
+            private final Map<String, String> appProperties = new HashMap<>();
+            private final Set<Operation> customOps = new HashSet<>();
+
+            /**
+             * Set the required {@link State} field.
+             */
+            public Builder state(@NonNull final State state) {
+                this.state = requireNonNull(state);
+                return this;
+            }
+
+            /**
+             * Add or override a single property.
+             */
+            public Builder appProperty(@NonNull final String key, @NonNull final String value) {
+                requireNonNull(key);
+                requireNonNull(value);
+                this.appProperties.put(key, value);
+                return this;
+            }
+
+            /**
+             * Add/override multiple properties at once.
+             */
+            public Builder appProperties(@NonNull final Map<String, String> properties) {
+                requireNonNull(properties);
+                this.appProperties.putAll(properties);
+                return this;
+            }
+
+            /**
+             * Set the optional {@link TracerBinding}.
+             */
+            public Builder customTracerBinding(@Nullable final TracerBinding customTracerBinding) {
+                this.customTracerBinding = customTracerBinding;
+                return this;
+            }
+
+            /**
+             * Set the custom operations in bulk.
+             */
+            public Builder customOps(@NonNull final Set<? extends Operation> customOps) {
+                requireNonNull(customOps);
+                this.customOps.addAll(customOps);
+                return this;
+            }
+
+            /**
+             * Add a single custom operation.
+             */
+            public Builder addCustomOp(@NonNull final Operation customOp) {
```

Comment:
Name `addCustomOp` doesn't seem to match pattern in this builder; as opposed to `customOp`, say.

Reply:
Hmm agreed, will change next time I touch this. 💯